drivers/regex/mlx5/mlx5_rxp.c

```diff
@@ -1157,7 +1154,7 @@ mlx5_regex_rules_db_import(struct rte_regexdev *dev,
 		return ret;
 	}
 	/* Need to ensure RXP not busy before stop! */
-	for (id = 0; id < MLX5_RXP_MAX_ENGINES; id++) {
+	for (id = 0; id < priv->num_engines; id++) {
```

Comment:
I suggest only here to have priv->num_engines.
This is the only place you actually talk to FW, right?

Otherwise, what the point of having MLX5_RXP_EM_COUNT=2?